backend/src/ml_space_lambda/dataset/lambda_functions.py

```diff
@@ -259,6 +294,12 @@ def create_dataset(event, context):
         if dataset_created:
             dataset_dao.delete(dataset_scope, dataset_name)
```

Comment:
Would we want to clean up group_datasets before deleting the dataset? This isn't relational so it doesn't matter but might make more sense

Reply:
I don't think it matters either way 🤷🏻‍♂️ I was just trying to maintain the order of

1. we create the dataset
2. we create the group_dataset rows

if things go wrong
1. we delete the dataset
2. we delete the group_dataset rows